src/cascadia/TerminalSettingsModel/ActionArgs.cpp

```diff
@@ -76,6 +77,53 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         return winrt::hstring{ s.substr(0, s.size() - 2) };
     }
 
+    winrt::hstring NewTerminalArgs::ToCommandline() const
+    {
+        std::wstringstream ss;
+
+        if (!_Profile.empty())
+        {
+            ss << fmt::format(L"--profile \"{}\" ", _Profile);
+        }
+        // The caller is always expected to provide the evaluated profile in the
+        // NewTerminalArgs, not the index
+        //
+        // else if (_ProfileIndex)
+        // {
+        //     ss << fmt::format(L"profile index: {}, ", _ProfileIndex.Value());
+        // }
+
+        if (!_StartingDirectory.empty())
+        {
+            ss << fmt::format(L"--startingDirectory \"{}\" ", _StartingDirectory);
+        }
+
+        if (!_TabTitle.empty())
+        {
+            ss << fmt::format(L"--title \"{}\" ", _TabTitle);
+        }
+
+        if (_TabColor)
+        {
+            const til::color tabColor{ _TabColor.Value() };
+            ss << fmt::format(L"--tabColor \"{}\" ", tabColor.ToHexString(true));
+        }
+
+        if (!_Commandline.empty())
+        {
+            ss << fmt::format(L"-- \"{}\" ", _Commandline);

```

Comment:
i think there's a better way to do this (using format_to), but i am not blocking on it